plugins/module_utils/aci.py

```diff
@@ -1472,26 +1472,30 @@ def payload(self, aci_class, class_config, child_configs=None):
 
         # add child objects to proposed
         if child_configs:
-            children = []
-            for child in child_configs:
-                child_copy = deepcopy(child)
-                has_value = False
-                for root_key in child_copy.keys():
-                    for final_keys, values in child_copy[root_key]["attributes"].items():
-                        if values is None:
-                            child[root_key]["attributes"].pop(final_keys)
-                        else:
-                            child[root_key]["attributes"][final_keys] = str(values)
-                            has_value = True
-                    # TODO: This needs to be recursive in order to handle attributes and the children of the subsequent children of the child
-                    if child_copy[root_key].get("children") is not None:
-                        has_value = True
-                if has_value:
-                    children.append(child)
+            children = self.handle_child_configs(child_configs)
 
             if children:
                 self.proposed[aci_class].update(dict(children=children))
 
+    def handle_child_configs(self, child_configs):
+        children = []
+        for child in child_configs:
+            child_copy = deepcopy(child)
+            has_value = False
+            for root_key in child_copy.keys():
+                for final_keys, values in child_copy[root_key]["attributes"].items():
```

Comment:
```suggestion
                for attributes, values in child_copy[root_key]["attributes"].items():
```
I think this is clearer on what this key is.
Should it be singular and not plural?

Reply:
I think its singular but this was part of the original code..